src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -167,4 +167,23 @@ protected function processValue(mixed $value, bool $isRoot = false): mixed
 
         return parent::processValue($value, $isRoot);
     }
+
+    /**
+     * Looks for a configurator for the given attribute name in the given configurators array.
+     * If no configurator is found, the parent class is checked.
+     * If no configurator is found, null is returned.
+     * The result is cached in the configurators array passed by reference.
+     */
+    private function findConfigurator(array &$configurators, string $attributeName): ?callable
+    {
+        if (\array_key_exists($attributeName, $configurators)) {
+            return $configurators[$attributeName];
+        }
+
+        if (class_exists($attributeName) && $parent = get_parent_class($attributeName)) {
```

Comment:
Don't you want a loop here to cascade the inheritance chain?

Reply:
I chose to cache the result in the `$configurators` array. If I do a loop, I need to store the list of classes names to be set in this array. I can to do this, the code will be a bit more complex but perhaps more efficient.